tests/data_context/test_data_context.py

```diff
@@ -1935,3 +1935,202 @@ def test_get_batch_multiple_datasources_do_not_scan_all(
         expectation_suite_name=expectation_suite,
     )
     assert len(batch) == 3
+
+
+def test_add_checkpoint_from_yaml(empty_data_context):
+    """
+    What does this test and why?
+    We should be able to add a checkpoint directly from a valid yaml configuration.
+    test_yaml_config() should not automatically save a checkpoint if valid.
+    checkpoint yaml in a store should match the configuration, even if created from SimpleCheckpoints
+    Note: This tests multiple items and could stand to be broken up.
+    """
+
+    context: DataContext = empty_data_context
+    checkpoint_name: str = "my_new_checkpoint"
+
+    assert checkpoint_name not in context.list_checkpoints()
+    assert len(context.list_checkpoints()) == 0
+
+    checkpoint_yaml_config = f"""
+name: {checkpoint_name}
+config_version: 1.0
+class_name: SimpleCheckpoint
+run_name_template: "%Y%m%d-%H%M%S-my-run-name-template"
+validations:
+  - batch_request:
+      datasource_name: data_dir
+      data_connector_name: data_dir_example_data_connector
+      data_asset_name: DEFAULT_ASSET_NAME
+      partition_request:
+        index: -1
+    expectation_suite_name: newsuite
+    """
+
+    checkpoint_from_test_yaml_config = context.test_yaml_config(
+        checkpoint_yaml_config, name=checkpoint_name
+    )
+
+    # test_yaml_config() no longer stores checkpoints automatically
+    assert checkpoint_name not in context.list_checkpoints()
+    assert len(context.list_checkpoints()) == 0
+
+    checkpoint_from_yaml = context.add_checkpoint(
+        **yaml.load(checkpoint_yaml_config),
+    )
+
+    expected_checkpoint_yaml: str = """name: my_new_checkpoint
+config_version: 1.0
+template_name:
+module_name: great_expectations.checkpoint
+class_name: Checkpoint
```

Comment:
This functionality of a user saving a `SimpleCheckpoint` and getting back a `Checkpoint` makes sense to me for a number of reasons, and I'd like to make sure it makes sense to @roblim and @alexsherstinsky!

Reply:
@Aylr: It makes sense, because while the config is that of the `SimpleCheckpoint`, the object is a `Checkpoint`, right?